src/runtime/converter.cs

```diff
@@ -133,6 +134,22 @@ internal static IntPtr ToPython(object value, Type type)
                 return result;
             }
 
+            if (value is IList && value.GetType().IsGenericType)
+            {
```

Comment:
To be completely consistent with the converter code the IEnumerable should be replaced with IList like you did this pull request.

The commit message mentioned that strings should not be converted to `PyList`. 

https://github.com/pythonnet/pythonnet/commit/471673a1fa4691b3becb4d9fee6941308535bb04

But `IEnumerable` is not quite right when converting to `PyList` anyway. `IList` is the closest interface like you selected. 
Other example which are IEnumerable, but not IList besides strings are HashSet, maybe even HashTable.